anchore_engine/services/policy_engine/engine/feeds/feeds.py

```diff
@@ -733,6 +734,53 @@ class GithubFeed(VulnerabilityFeed):
     )
 
 
+class MicrosoftFeed(AnchoreServiceFeed):  # update this to VulnerabilityFeed since its the same
+    """
+    Feed for Microsoft data served from on-prem enterprise feed service
+    """
+
+    __feed_name__ = 'microsoft'
+    _products_group = 'msrc:products'
+    _cve_key = 'id'
+    __group_data_mappers__ = MultipleTypeMapperFactory() \
+        .add(group=_products_group, single_type_mapper_factory=SingleTypeMapperFactory(__feed_name__, MSRCProductFeedDataMapper, _cve_key)) \
+        .add(group='msrc:.+', single_type_mapper_factory=SingleTypeMapperFactory(__feed_name__, MSRCVulnerabilityFeedDataMapper, _cve_key))
+
+    def _flush_group(self, group_obj, operation_id=None):
+        log.info(log_msg_ctx(operation_id, group_obj.name, group_obj.feed_name, 'Flushing group records'))
+
+        db = get_session()
+
+        if group_obj.name == MicrosoftFeed._products_group:
+            count = db.query(GenericFeedDataRecord).filter(GenericFeedDataRecord.group == group_obj.name).delete()
+            log.info(log_msg_ctx(operation_id, group_obj.name, group_obj.feed_name, 'Flushed {} group records'.format(count)))
+            group_obj.last_sync = None  # Null the update timestamp to reflect the flush
+            group_obj.count = 0
+        else:
+            # VulnerabilityFeed.__flush_helper_fn__(db=db, feed_name=group_obj.feed_name, group_name=group_obj.name)
+
+            count = db.query(FixedArtifact).filter(FixedArtifact.namespace_name == group_obj.name).delete()
```

Comment:
These shouldn't be necessary, the relationship cascades will handle it. Only the Vulnerability object delete is needed to handle both artifact flushes.

Reply:
Fixed